espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/KReflectionUtils.kt

```diff
@@ -0,0 +1,67 @@
+package io.appium.espressoserver.lib.helpers
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException
+import java.lang.IllegalArgumentException
+import java.lang.reflect.InvocationTargetException
+import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
+import kotlin.reflect.full.createType
+import kotlin.reflect.full.functions
+import kotlin.reflect.full.memberFunctions
+import kotlin.reflect.jvm.javaType
+
+object KReflectionUtils {
+
+    fun invokeMethod(functions: Collection<KFunction<*>>, methodName: String, vararg providedParams: Any): Any? {
+        val treatedParams = providedParams.clone().toMutableList()
+        for (func in functions) {
+            // Look for function names that match provided methodName
+            if (func.name == methodName) {
+
+                // If the length of func parameters provided isn't same as expected, go to next.
+                val funcParams = func.parameters
+                if (funcParams.size != providedParams.size) {
+                    continue
+                }
+
+                // Look through function parameters and do an enum hack to translate strings to enums
+                for (index in 0 until funcParams.size) {
+                    val funcParamType = funcParams.get(index).type
+                    val providedParam = providedParams.get(index)
+
+                    // Hack Enum Case
+                    // If function param is Enum and provided param is String, try `enumValueOf` on that String value
+                    try {
+                        val jFuncType = funcParamType.javaType as Class<*>
+                        if (jFuncType.isEnum && providedParam is String) {
+                            val enumValueOf = jFuncType.getDeclaredMethod("valueOf", String::class.java)
+                            treatedParams.set(index, enumValueOf(null, providedParam.toUpperCase()))
+                            continue;
+                        }
+                    } catch (e:Exception) {
```

Comment:
do we have more precise exception class?

Reply:
Yeah, I'm going to change it to the broader `ReflectiveTypeException` class. That should cover it.